docs/react-node.md

```diff
@@ -0,0 +1,13 @@
+---
+id: react-node
```

Comment:
I feel like if this is documented it should be in the React repo and not the React Native repo. This is anything that React allows as a child. 

Is there an existing page that React.Node can link to?

Reply:
I also got hung up on this! But talked with Dan about it, and we are following his advice verbatim with these pages. (React node is more of a typing construct than something React exports, after all.)